tests/spec/fs.watchFile.spec.js

```diff
@@ -8,24 +8,27 @@ describe('fs.watchFile', function() {
   afterEach(util.cleanup);
 
   it('should be a function', function() {
-    var fs = util.fs();
+    const fs = util.fs();
     expect(typeof fs.watchFile).to.equal('function');
   });
 
-  /*
-  it('should get a change event when writing a file', function(done) {
+  it('should throw an error if a file path is not defined', function() { 
     const fs = util.fs();
+    
+    const fn = () => fs.watchFile(undefined);
+    expect(fn).to.throw();
+  });
 
-    fs.watchFile('/myfile.txt', function(event, filename) {
-      expect(event).to.equal('change');
-      expect(filename).to.equal('/myfile.txt');
-      watcher.close();
-      done();
-    });
+  it('prev and curr should be populated', function() {
+    const fs = util.fs(); 
 
-    fs.writeFile('/myfile.txt', 'data', function(error) {
+    fs.writeFile('/myfile', 'data', function(error) {
       if(error) throw error;
+    }); 
+    
+    fs.watchFile('/myfile', function(prev, curr) {
+      expect(prev).to.exist;
+      expect(curr).to.exist;
```

Comment:
Let's do more than just check if the objects exist.  We'd expect them to exist, and also to match what a `stat` object looks like, see https://github.com/filerjs/filer/blob/master/src/stats.js#L10.

Some other tests that I can think of:

- if you `watchFile` a file, and then write to it, your `listener` should get called
- We'd want to see that the `stats` (`prev` vs. `curr`) do in fact change as the file changes.
- What happens if we delete a file?
- Does watchFile work across symlinks?
- Is the `interval` value passed by the user honoured in the tests?  What if they set another value, do we stick to that timing?

There are more we could do, but at the very least, we need proof that this is working as expected.

Reply:
I've worked on creating some tests for half the points mentioned, but I'm going to have to figure out the logic to test the ones I haven't covered.